Tools/CISupport/ews-build/events.py

```diff
@@ -278,6 +280,63 @@ def __exit__(self, type, value, tb):
         log.msg = self.saved_logger
 
 
+# Based off of webkitscmpy's CommitClassifier, although modified to not raise exceptions
+# when provided invalid commit classes.
+class CommitClass(object):
+    class HeaderFilter(object):
+        DEFAULT_FUZZ_RATIO = 90
+
+        @classmethod
+        def fuzzy(cls, string, ratio=None):
+            ratio = cls.DEFAULT_FUZZ_RATIO if not ratio else ratio
+            return lambda x: fuzz.partial_ratio(string, x) >= ratio
+
+        def __init__(self, value):
+            self.description = value
+            if isinstance(value, str):
+                self.do = lambda x: re.search(value, x)
+            elif isinstance(value, dict) and 'value' in value:
+                self.description = 'fuzz({}, {}%)'.format(value['value'], value.get('ratio', self.DEFAULT_FUZZ_RATIO))
+                self.do = self.fuzzy(value['value'], ratio=value.get('ratio'))
+            else:
+                self.do = None
+
+        def __repr__(self):
+            return self.description
+
+        def __call__(self, string):
+            return bool(self.do(string)) if self.do else False
+
+    def __init__(self, name=None, pickable=True, headers=None, paths=None, **kwargs):
+        self.name = name or '?'
+        self.pickable = pickable
+        self.headers = [self.HeaderFilter(header) for header in headers or []]
+        self.paths = [re.compile(r'^{}'.format(path)) for path in (paths or [])]
+
+        if not self.headers and not self.paths:
+            log.msg(f'Commit class {self.name} matches all commits')
+
+        for argument, _ in kwargs.items():
+            log.msg(f'{argument} is not a valid member of CommitClass')
+
+    def matches(self, header, files):
+        if not self.headers and not self.paths:
+            return False
+
+        if self.headers and not header:
+            return False
+        if self.headers and not any([h(header) for h in self.headers]):
```

Comment:
can `h(header)` raise exception (is `h` always callable)?  If so, might consider exception handling here.